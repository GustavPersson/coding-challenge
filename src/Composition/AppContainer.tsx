import { Container } from "inversify";
import SearchViewModel from "../pages/search/SearchViewModel";
import ApiService from "../service/ApiService";
import FormatList from "../service/FormatList";
import ArticleViewModel from "../pages/article/ArticleViewModel";

export type Containers =
  | "API_SERVICE"
  | "FORMAT_LIST_SERVICE"
  | "SEARCH_PAGE"
  | "ARTICLE_PAGE";

const container = new Container();

container.bind("API_SERVICE").to(ApiService).inSingletonScope();
container.bind("FORMAT_LIST_SERVICE").to(FormatList).inSingletonScope();
container.bind("SEARCH_PAGE").to(SearchViewModel).inSingletonScope();
container.bind("ARTICLE_PAGE").to(ArticleViewModel).inSingletonScope();

export const appContainer: Container = container;
