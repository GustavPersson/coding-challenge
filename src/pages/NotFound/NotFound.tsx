import { ReactElement, useEffect } from "react";
import { Link } from "react-router-dom";
import SectionDescription from "../../components/SectionDescription";
import SectionHeader from "../../components/SectionHeader";
import SectionStack from "../../components/SectionStack";

function NotFound({
  setTitle,
}: {
  setTitle: (title: string) => void;
}): ReactElement {
  useEffect(() => {
    setTitle("Page not Found");
  }, []);

  return (
    <SectionStack>
      <SectionHeader>Page not found</SectionHeader>
      <SectionDescription>
        Sorry! You seem to have navigated to a page that does not exist, oh no!
      </SectionDescription>
      <Link to="search">Go to the Search Page</Link>
    </SectionStack>
  );
}

export default NotFound;
