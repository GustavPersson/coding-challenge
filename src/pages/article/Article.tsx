import {
  Box,
  Grid,
  Stack,
  Typography,
  CircularProgress,
  ImageList,
  ImageListItem,
  Chip,
  Rating,
  TableContainer,
  Table,
  TableBody,
  TableCell,
  TableRow,
} from "@mui/material";
import { Observer } from "mobx-react-lite";
import { SyntheticEvent, useEffect, useState } from "react";
import useContainer from "../../utils/useContainer";
import { ArticleViewModelType } from "./ArticleViewModel";
import { useNavigate, useParams } from "react-router-dom";

type Params = {
  id: string;
};

function SearchResultsPage({
  setTitle,
}: {
  setTitle: (title: string) => void;
}) {
  const [loading, setLoading] = useState(true);
  const params = useParams<Params>();
  const viewModel = useContainer<ArticleViewModelType>("ARTICLE_PAGE");
  const navigate = useNavigate();

  useEffect(() => {
    setLoading(true);
    const getArticle = async () => {
      if (!params.id || !Number.isInteger(+params.id)) {
        navigate("/404", { replace: true });
        return;
      }
      try {
        await viewModel.getArticle(+params.id);
        setTitle(viewModel.article.title);
      } catch (error) {
        navigate("/404", { replace: true });
        return;
      }
      setLoading(false);
    };
    getArticle();
  }, [params.id]);

  const onImageError = (
    ev: SyntheticEvent<HTMLImageElement, Event>,
    height: number,
    width: number
  ) => {
    (
      ev.target as HTMLImageElement
    ).src = `https://placehold.co/${height}x${width}`;
  };

  if (loading) {
    return (
      <Box>
        <Stack direction={"row"} justifyContent={"center"}>
          <CircularProgress />
        </Stack>
      </Box>
    );
  }

  return (
    <Observer>
      {() => (
        <Grid container spacing={2}>
          <Grid item container xs={12}>
            <Grid item container spacing={2} flexWrap={"nowrap"}>
              <Grid item xs="auto">
                <img
                  src={viewModel.article.thumbnail}
                  onError={(ev) => onImageError(ev, 45, 45)}
                  height={45}
                  width={45}
                />
              </Grid>
              <Grid item>
                <Typography data-testid="article-title" variant="h5">
                  {viewModel.article.title}
                </Typography>
                <Typography>{viewModel.article.description}</Typography>
              </Grid>
            </Grid>
          </Grid>
          <Grid item container xs={12} spacing={1}>
            <Grid item>
              <Chip label={viewModel.article.brand} variant="outlined" />
            </Grid>
            <Grid item>
              <Chip label={viewModel.article.category} variant="outlined" />
            </Grid>
            {viewModel.article.stock > 0 && (
              <Grid item>
                <Chip label="In Stock" variant="outlined" color="success" />
              </Grid>
            )}
            {viewModel.article.stock === 0 && (
              <Grid item>
                <Chip label="Out of stock" variant="outlined" color="error" />
              </Grid>
            )}
          </Grid>
          <Grid item xs={12}>
            <Box>
              <Typography component="legend">Rating</Typography>
              <Rating
                name="rating"
                value={viewModel.article.rating}
                precision={0.5}
                readOnly
              />
            </Box>
          </Grid>
          <Grid item xs={12}>
            <ImageList>
              {viewModel.article.images.map((image) => (
                <ImageListItem key={image}>
                  <img
                    src={image}
                    onError={(ev) => onImageError(ev, 164, 164)}
                    loading="lazy"
                  />
                </ImageListItem>
              ))}
            </ImageList>
          </Grid>
          <Grid item xs={12}>
            <TableContainer>
              <Table>
                <TableBody>
                  {viewModel.getTableRows.map((row) => (
                    <TableRow
                      key={row.name}
                      sx={{ "&:last-child td, &:last-child th": { border: 0 } }}
                    >
                      <TableCell component="th" scope="row">
                        {row.name}
                      </TableCell>
                      <TableCell align="right">
                        {row.value}
                        {row.unit}
                      </TableCell>
                    </TableRow>
                  ))}
                </TableBody>
              </Table>
            </TableContainer>
          </Grid>
        </Grid>
      )}
    </Observer>
  );
}

export default SearchResultsPage;
