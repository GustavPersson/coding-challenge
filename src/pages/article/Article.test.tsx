import { render, waitFor, screen } from "@testing-library/react";
import Article from "./Article";
import {
  MemoryRouter,
  Route,
  RouterProvider,
  Routes,
  createMemoryRouter,
} from "react-router-dom";
import NotFound from "../NotFound/NotFound";
import { appContainer } from "../../Composition/AppContainer";

describe("ArticlePage", () => {
  const setTitle = jest.fn();
  const routes = [
    {
      path: "/article/:id",
      element: <Article setTitle={setTitle} />,
    },
    {
      path: "*",
      element: <NotFound setTitle={setTitle} />,
    },
  ];

  beforeEach(() => {
    appContainer.snapshot();
    setTitle.mockReset();
  });

  afterEach(() => {
    appContainer.restore();
  });

  it("should render", async () => {
    const router = createMemoryRouter(routes, {
      initialEntries: ["/article/16"],
    });

    const { container } = render(<RouterProvider router={router} />);

    await waitFor(() => screen.getByText("Hyaluronic Acid Serum"));
    expect(container).toMatchSnapshot();
    expect(setTitle).toBeCalledWith("Hyaluronic Acid Serum");
  });

  it("should navigate to not found on missing params", async () => {
    const router = createMemoryRouter(routes, {
      initialEntries: ["/article"],
    });
    const { container } = render(<RouterProvider router={router} />);
    await waitFor(() => screen.getByText("Page not found"));
    expect(container).toMatchSnapshot();
    expect(setTitle).toBeCalledWith("Page not Found");
  });

  it("should navigate to not found on incorrect param", async () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/article/blorg"]}>
        <Routes>
          <Route
            path="/article/:id"
            element={<Article setTitle={setTitle} />}
          />

          <Route path="/*" element={<NotFound setTitle={setTitle} />} />
        </Routes>
      </MemoryRouter>
    );

    await waitFor(() => screen.getByText("Page not found"));
    expect(container).toMatchSnapshot();
    expect(setTitle).toBeCalledWith("Page not Found");
  });

  it("should navigate to not found on article not found", async () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/article/1092"]}>
        <Routes>
          <Route
            path="/article/:id"
            element={<Article setTitle={setTitle} />}
          />

          <Route path="/*" element={<NotFound setTitle={setTitle} />} />
        </Routes>
      </MemoryRouter>
    );
    await waitFor(() => screen.getByText("Page not found"));
    expect(container).toMatchSnapshot();
    expect(setTitle).toBeCalledWith("Page not Found");
  });
});
