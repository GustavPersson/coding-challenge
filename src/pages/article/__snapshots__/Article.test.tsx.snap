// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePage should navigate to not found on article not found 1`] = `
<div>
  <div
    class="css-1lwbda4-MuiStack-root"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
    >
      Page not found
    </h3>
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Sorry! You seem to have navigated to a page that does not exist, oh no!
    </p>
    <a
      href="/search"
    >
      Go to the Search Page
    </a>
  </div>
</div>
`;

exports[`ArticlePage should navigate to not found on incorrect param 1`] = `
<div>
  <div
    class="css-1lwbda4-MuiStack-root"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
    >
      Page not found
    </h3>
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Sorry! You seem to have navigated to a page that does not exist, oh no!
    </p>
    <a
      href="/search"
    >
      Go to the Search Page
    </a>
  </div>
</div>
`;

exports[`ArticlePage should navigate to not found on missing params 1`] = `
<div>
  <div
    class="css-1lwbda4-MuiStack-root"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
    >
      Page not found
    </h3>
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Sorry! You seem to have navigated to a page that does not exist, oh no!
    </p>
    <a
      href="/search"
    >
      Go to the Search Page
    </a>
  </div>
</div>
`;

exports[`ArticlePage should render 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-spompt-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 css-1h2b65z-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
        >
          <img
            height="45"
            src="/products/16/thumbnail.jpg"
            width="45"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            data-testid="article-title"
          >
            Hyaluronic Acid Serum
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            L'OrÃ©al Paris introduces Hyaluron Expert Replumping Serum formulated with 1.5% Hyaluronic Acid
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 MuiGrid-grid-xs-12 css-bzgugk-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1012bww-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            L'Oreal Paris
          </span>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1012bww-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            skincare
          </span>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorSuccess MuiChip-outlinedSuccess css-gjv9n6-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            In Stock
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-0"
      >
        <legend
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Rating
        </legend>
        <span
          aria-label="5 Stars"
          class="MuiRating-root MuiRating-sizeMedium MuiRating-readyOnly css-ryrseu-MuiRating-root"
          role="img"
        >
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
        </span>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <ul
        class="MuiImageList-root MuiImageList-standard css-rvv586-MuiImageList-root"
        style="grid-template-columns: repeat(2, 1fr); gap: 4px;"
      >
        <li
          class="MuiImageListItem-root MuiImageListItem-standard css-ta95xa-MuiImageListItem-root"
          style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
        >
          <img
            class="MuiImageListItem-img"
            loading="lazy"
            src="/products/16/1.png"
          />
        </li>
        <li
          class="MuiImageListItem-root MuiImageListItem-standard css-ta95xa-MuiImageListItem-root"
          style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
        >
          <img
            class="MuiImageListItem-img"
            loading="lazy"
            src="/products/16/2.webp"
          />
        </li>
        <li
          class="MuiImageListItem-root MuiImageListItem-standard css-ta95xa-MuiImageListItem-root"
          style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
        >
          <img
            class="MuiImageListItem-img"
            loading="lazy"
            src="/products/16/3.jpg"
          />
        </li>
        <li
          class="MuiImageListItem-root MuiImageListItem-standard css-ta95xa-MuiImageListItem-root"
          style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
        >
          <img
            class="MuiImageListItem-img"
            loading="lazy"
            src="/products/16/4.jpg"
          />
        </li>
        <li
          class="MuiImageListItem-root MuiImageListItem-standard css-ta95xa-MuiImageListItem-root"
          style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
        >
          <img
            class="MuiImageListItem-img"
            loading="lazy"
            src="/products/16/thumbnail.jpg"
          />
        </li>
      </ul>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-rqglhn-MuiTable-root"
        >
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-34nofg-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                scope="row"
              >
                Price
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
              >
                19
                kr
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-34nofg-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                scope="row"
              >
                Stock
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
              >
                110
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-34nofg-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                scope="row"
              >
                Discount
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
              >
                13.31
                %
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
